app/controllers/course/learning_map_controller.rb

```diff
@@ -1,15 +1,219 @@
 # frozen_string_literal: true
 class Course::LearningMapController < Course::ComponentController
+  NODE_ID_DELIMITER = '-'
+  NEGATIVE_INF = -1_000_000_000
+
+  before_action :authorize_update, only: [:add_parent_node, :remove_parent_node, :toggle_satisfiability_type]
   add_breadcrumb :index, :course_learning_map_path
 
   def index
+    init_conditionals
```

Comment:
I think in a subsequent PR, you can check that the user has access to these conditionals.

Reply:
Since `index` isn't guarded by any authorisation currently.